### DISCUSSION TITLE
Process interactive bug on hover

### DISCUSSION THREAD

**Discussion by sarahgaucimizzi** [pixijs/pixijs#7680] [General]
After updating to 6.1, it seems that the hit points test are returning this. 
Nothing was changed from previous version using 5.3.4

### Expected Behavior
No errors

### Current Behavior
Gives this error:
![image](https://user-images.githubusercontent.com/11066256/128334475-cb419212-c507-49b5-9e34-f614e481646d.png)

### Steps to Reproduce
Hover over any Container

### Environment
Webpack

- **`pixi.js` version**: _6.1.0_
- **Browser & Version**: _Chrome Version 92.0.4515.131 (Official Build) (x86_64)_
- **OS & Version**: _MacOs 11.4_

**Comment by ivanpopelyshev** [comment]
two versions of pixi in node_modules.
rm package.lock
rm node_modules -rf
npm i

**Comment by sarahgaucimizzi** [comment]
Tried that but still did not solve the issue.

**Comment by bigtimebuddy** [comment]
Could you create a minimal reproduction of this issue using codepen, jsfiddle etc? That would help us a lot verify and fix the issue.

**Comment by sarahgaucimizzi** [comment]
It did not give any issues recreating it from scratch. 

We use an architecture where game clients use a common library and they both install pixi.js but then they are being resolved through webpack resolve alias to use only one. And all was working fine with version 5.3.4.

We are trying to investigate the issues ourselves too, but we thought you might have better insight.

**Comment by bigtimebuddy** [comment]
If you identify the underlying issue, please keep us posted. Nothing comes to mind immediately. There were, I don't think, many significant changes to interaction in 6.1.0.

**Comment by bigtimebuddy** [comment]
Also please update to 6.0.4 to see if that makes a difference and isolate if it's just a 6.1.0 issue.

**Comment by MercurialUroboros** [comment]
I'm a colleague of @sarahmizzi. I tried to downgrade to 6.0.4 and the issue is still there. 
Gonna try to downgrade one by one till 6.0.0.

**Comment by MercurialUroboros** [comment]
Found the issue. 
We tried to use https://preview.npmjs.com/package/@pixi/graphics-extras for a .drawStar.
(Not only the methods of the graphics extra won't work as expected- Cannot invoke an object which is possibly 'undefined')
But on top of that, adding event listeners makes that happen.
Will now try to debug even further.

Discovery 1: 
The "Graphics" just need to have interactive = true to make this hell happens.

Replacing with just a normal graphics will make It work, nothing else found.

**Comment by ivanpopelyshev** [comment]
Here's mega snippet from GS examples: `GraphicsSmooth.prototype.drawStar =Graphics prototype.drawStar`

**Comment by MercurialUroboros** [comment]
Really I just tried to follow the guide on graphics-extra so not sure what's that snippet supposed to do. 
Also, we found a way around using it as it wasn't crucial to have, so "Problem Solved".

**Comment by ivanpopelyshev** [comment]
> not sure what's that snippet supposed to do.
https://github.com/pixijs/pixijs/blob/01a69a1465b875cb7b97a10caa9c35d9505397ff/packages/graphics-extras/src/index.ts#L25
https://github.com/pixijs/pixijs/blob/01a69a1465b875cb7b97a10caa9c35d9505397ff/packages/graphics-extras/src/drawStar.ts#L65

Its using higher API of Graphics to draw things. GraphicsSmooth has same higher API, and I thought it would be obvious for users how to add those methods inside. Gonna add it in README because of you :)

**Comment by MercurialUroboros** [comment]
It could be beneficial to add some docs on the graphics-extra library as I really just tried what's in that piece of documentation, didn't really dig in the code as I don't have much time for finding stuff. 
(Considering that drawStar was just part of Graphics, and now it's spread amongst 2 more libraries, GraphicsSmooth And GraphicsExtra, that I didn't really knew the existence)

**Comment by MercurialUroboros** [comment]
Anyway, the issue seems to be bound to our codebase, not reproducible on a clean project. I believe it boils down to package resolve through webpack, thanks for the help guys.

**Comment by ivanpopelyshev** [comment]
>  I don't have much time for finding stuff

Here's my honest warning: effectively using pixi or any other WebGL lib requires you to clone them and open in separate IDE window, so you can just look for classes/files that you use (learn the shortcuts!). Otherwise, any problem like that will eat significant portion of your time.

Especially because its in active development and code is getting shuffled around by several people.

That's what we all have to pay for using WebGL :(

> I believe it boils down to package resolve through webpack

Do you, by any chance, use `pixi-spine` ? We had a bit of depts problem there that can affect pixi modules resolution

**Comment by MercurialUroboros** [comment]
> Here's my honest warning: effectively using pixi or any other WebGL lib requires you to clone them and open in separate IDE window, so you can just look for classes/files that you use (learn the shortcuts!). Otherwise, any problem like that will eat significant portion of your time.

I thought so,  saw many posts and usually, it resolves through investigation of code.
Normally I would but I'm having tight deadlines and I tried from docs.
Thanks for the clarification.



> Do you, by any chance, use `pixi-spine` ? We had a bit of depts problem there that can affect pixi modules resolution

Yes, I was using 3.0.7 when that happened.

**Comment by ivanpopelyshev** [comment]
Update to 3.0.8, i fixed depts and some other things :)

**Comment by MercurialUroboros** [comment]
Tried. On a clean project either I use 3.0.7 or 3.0.8 doesn't show any issue.  ✅ 
On the main codebase 3.0.8 doesn't solve the issue. ❌

**Comment by ivanpopelyshev** [comment]
> On the main codebase 3.0.8 doesn't solve the issue. 

`rm package.lock` , `rm node_modules -rf`, `npm i` didnt do the job? :(

I'm sorry you have to go through that deps nonsense, as you see we figured out correct approach 2 years after we released first attempt in pixi-v5. 3 years since we started develop it.

**Comment by MercurialUroboros** [comment]
I think I did, let me try one more time just in case.

**Comment by MercurialUroboros** [comment]
Yup, still the same issue. I'm gonna try later in the day to resolve through webpack the graphics package, cause it might be conflicting.

**Comment by ivanpopelyshev** [comment]
Just checked graphics-smooth - it uses `"^6.0.4"` so it should be satisfied with `6.1`. Tell me what other plugins do you use?

**Reply by MercurialUroboros** [reply]
```
    "pixi-particles": "^4.3.1",
    "pixi-spine": "^3.0.8",
    "pixi.js": "^6.1.1",
    "@pixi/sound": "^4.0.4",
    "pixi-filters": "^4.1.4",
    "pixi-particles": "^4.3.1",
```

These are all the packages I'm currently using Pixi-related

**Comment by bigtimebuddy** [comment]
Because this issue has turned more into support, I'm converting it to a discussion. If there's a specific bug, please make a new issue.
